r/R/array.R

```diff
@@ -450,6 +440,44 @@ StructArray <- R6Class("StructArray",
   )
 )
 
+StructArray$create <- function(...) {
+  dots <- list2(...)
```

Comment:
I'm not sure how to make that work; I tried start off with:

```
dots <- list2(...)

stopifnot(length(dots) > 0)

if (is.null(names(dots))) {
  names(dots) <- rep_len("", length(dots))
}

batch <- record_batch(dots)
```

but I just end up with

`Error: Unknown: only data frames are allowed as unnamed arguments to be auto spliced`

Reply:
Perhaps the solution is just to have some non-empty default names? perhaps `x1, x2, x3, ...`?

Reply:
That doesn't work as it adds names to things which shouldn't be named like data.frames.

Reply:
Ah thanks, but solved it a little differently. I've ended up sticking with the previous non-record batch approach as I wanted to avoid creating additional R6 objects en route, and this approach seems to work now.